pkg/storage/cloud/external_storage.go

```diff
@@ -474,13 +474,120 @@ func (h *httpStorage) Conf() roachpb.ExternalStorage {
 	}
 }
 
+type reqSender = func(reqHeaders map[string]string) (io.ReadCloser, http.Header, error)
+
+type bodyReader struct {
```

Comment:
2¢: I might call this httpReader or resumingHTTPReader since there are other things in this package that might have bodies that can be read.

Reply:
done